lark/tree.py

```diff
@@ -109,17 +112,18 @@ def iter_subtrees(self) -> 'Iterator[Tree]':
         subtrees = OrderedDict()
         for subtree in queue:
             subtrees[id(subtree)] = subtree
-            queue += [c for c in reversed(subtree.children)
+            # Reason for type ignore https://github.com/python/mypy/issues/10999
```

Comment:
With the current type system it isn't possible to convey that  "`_Leaf_T` can be anything except a `Tree`". So this will likely always be a thing `mypy` won't be able to figure out on its own. The referenced issue has more context and might provide some insights on how to better handle it.